### PR TITLE
feat: clickable username in post summery card

### PR DESCRIPTION
**Pull request by awais-ansari**
### [INF-243](https://2u-internal.atlassian.net/browse/INF-243)
1. Click on post summery card username will redirect to the user all created and interacted posts under the learner tab.

### CODE REVIEW

File: src/discussions/learners/LearnerPostsView.jsx [hunk 1]
```diff
@@ -33,13 +33,14 @@ function LearnerPostsView({ intl }) {
   const { courseId, learnerUsername: username } = useContext(DiscussionContext);
   const nextPage = useSelector(selectThreadNextPage());
   const { id: userId } = useSelector(selectLearnerProfile(username));
+  const authorId = history.location.state?.authorId;
 
   useEffect(() => {
-    dispatch(fetchUserPosts(courseId, username, userId));
+    dispatch(fetchUserPosts(courseId, username, userId || authorId));
```
Comment (asadazam93) [inline src/discussions/learners/LearnerPostsView.jsx:39]: is there a case where the first userId will be null?

File: src/discussions/learners/LearnerPostsView.jsx [hunk 2]
```diff
@@ -88,7 +89,7 @@ function LearnerPostsView({ intl }) {
             <Spinner animation="border" variant="primary" size="lg" />
           </div>
         ) : (
-          nextPage && (
+          nextPage && loadingStatus === RequestStatus.SUCCESSFUL && (
```
Comment (asadazam93) [inline src/discussions/learners/LearnerPostsView.jsx:92]: is this change related? Or does this fix some other issue?

File: src/discussions/posts/post/PostLink.test.jsx [hunk 3]
```diff
@@ -15,7 +15,7 @@ function renderComponent(post) {
   return render(
     <IntlProvider locale="en">
       <AppProvider store={store}>
-        <PostLink post={post} key={post.id} isSelected={() => true} />
+        <PostLink post={post} key={post.id} isSelected={() => true} learnerTab />
```
Comment (mehaknasir) [inline src/discussions/posts/post/PostLink.test.jsx:18]: all test cases here assuming component called from learner tab? whereas this prop is not required to be passed in the component. why do we need that change in test cases?

File: src/discussions/learners/LearnerPostsView.jsx (hunk 2 above)
Comment (mehaknasir) [reply src/discussions/learners/LearnerPostsView.jsx:92]: this is an unrelated change as seen from PR description. I think it saves us from making extra call when loading new thread is already in progress.

**Review by mehaknasir — APPROVED** [review]
just a question posted, rest of the structure looks good to me. also if there is some fixes apart  from the actual ticket kindly add in PR description

**Comment by asadazam93** [comment]
@awais-ansari the PR requires rebase

File: src/discussions/posts/post/PostLink.test.jsx (hunk 3 above)
Comment (awais-ansari) [reply src/discussions/posts/post/PostLink.test.jsx:18]: Yes, you are right. I'll make it dynamic which will cover both cases leaner and simple posts.

**Review by asadazam93 — APPROVED** [review]
